src/pydargs/__init__.py

```diff
@@ -45,11 +45,31 @@ def parse(tp: Type[Dataclass], args: Optional[list[str]] = None, **kwargs: Any)
         An instance of tp.
     """
     namespace = _create_parser(tp, **kwargs).parse_args(args)
-    return tp(**namespace.__dict__)
+    return _create_object(tp, namespace)
+
+
+def _create_object(tp: Type[Dataclass], namespace: Namespace, prefix: str = "") -> Dataclass:
+    for field in fields(tp):
+        if hasattr(field.type, "__dataclass_fields__"):
```

Comment:
```suggestion
        if is_dataclass := hasattr(field.type, "__dataclass_fields__"):
```

An idea to indicate what we are checking here

Reply:
That isn't allowed by ruff;  ```F841 Local variable `is_dataclass` is assigned to but never used```

Reply:
Ah crap..